extensions/extension_mgr.go

```diff
@@ -20,45 +22,72 @@ type AdvancedExtension = extensions.AdvancedExtension
 
 const SubstraitDefaultURIPrefix = "https://github.com/substrait-io/substrait/blob/main/extensions/"
 
-// DefaultCollection is loaded with the default Substrait extension
-// definitions. Not all files are currently parsable.
-// Parser needs to enhanced to support all files
-var DefaultCollection Collection
+var (
+	defaultCollection   Collection
+	collectionOnce      sync.Once
+	collectionLoadError error
+)
+
+// GetDefaultCollection returns a Collection that is loaded with the default Substrait extension definitions.
+func GetDefaultCollection() *Collection {
+	c, err := GetDefaultCollectionWithError()
+	if err != nil {
+		panic(err)
+	}
+	return c
+}
+
+// GetDefaultCollectionWithError returns a Collection that is loaded with the default Substrait extension definitions.
+func GetDefaultCollectionWithError() (*Collection, error) {
+	collectionOnce.Do(func() {
```

Comment:
If you use sync.OnceValues you can have it return the data and the error together without needing to keep track of them separately (this obviates the need for all three private global variables).

Reply:
Thanks, the code is more simpler now.